OpenRA.Mods.Common/EditorBrushes/EditorTilingPathBrush.cs

```diff
@@ -0,0 +1,342 @@
+					var rallyIndex =
+						points
+							.Where(p => p.CPos == cpos)
+							.Select(p => p.RallyIndex)
+							.FirstOrDefault(0);
+					var isStartDirector =
+						plan.AutoStart != Direction.None
+							&& cpos == plan.FirstPoint - Direction.ToCVec(plan.AutoStart);
+					var isEndDirector =
+						plan.AutoEnd != Direction.None
+							&& cpos == plan.LastPoint + Direction.ToCVec(plan.AutoEnd);
+					return (isInside, isRally, rallyIndex, isStartDirector, isEndDirector);
+				}
+
+				var (fromIsInside, fromIsRally, fromRallyIndex, fromIsStartDirector, fromIsEndDirector) =
+					AssessCPos(from);
+				var (toIsInside, toIsRally, toRallyIndex, toIsStartDirector, toIsEndDirector) =
+					AssessCPos(to);
+
+				if (isDrag)
+				{
+					if (fromIsStartDirector)
+					{
+						var offset = plan.FirstPoint - to;
+						var direction =
+							offset != CVec.Zero
+								? Direction.FromCVecRounding(offset)
+								: Direction.None;
+						UpdatePlan(plan.WithStart(direction));
+					}
+					else if (fromIsEndDirector)
+					{
+						var offset = to - plan.LastPoint;
+						var direction =
+							offset != CVec.Zero
+								? Direction.FromCVecRounding(offset)
+								: Direction.None;
+						UpdatePlan(plan.WithEnd(direction));
+					}
+					else if (fromIsInside)
+					{
+						if (fromIsRally)
+						{
+							if (!toIsRally)
+							{
+								UpdatePlan(plan.WithRallyReplaced(fromRallyIndex, to));
+							}
+						}
+						else
+						{
+							UpdatePlan(plan.Moved(to - from));
+						}
+					}
+				}
+				else
+				{
+					if (toIsInside)
+					{
+						if (toIsRally)
+						{
+							if (toRallyIndex == 0)
+							{
+								UpdatePlan(plan.WithLoop(!plan.Loop));
+							}
+							else
+							{
+								UpdatePlan(plan.WithRallyRemoved(toRallyIndex));
+							}
+						}
+						else
+						{
+							UpdatePlan(plan.WithRallyInserted(toRallyIndex, to));
+						}
+					}
+					else
+					{
+						UpdatePlan(plan.WithRallyAppended(to));
+					}
+				}
+
+				return true;
+			}
+			else
+			{
+				return false;
+			}
+		}
+
+		void IEditorBrush.TickRender(WorldRenderer wr, Actor self) { }
+		IEnumerable<IRenderable> IEditorBrush.RenderAboveShroud(Actor self, WorldRenderer wr)
+		{
+			if (tool.EditorBlitSource == null)
+				yield break;
+
+			var preview = EditorBlit.PreviewBlitSource(
+				tool.EditorBlitSource.Value,
+				MapBlitFilters.Terrain | MapBlitFilters.Actors,
+				CVec.Zero,
+				wr);
+			foreach (var renderable in preview)
+				yield return renderable;
+		}
+
+		IEnumerable<IRenderable> IEditorBrush.RenderAnnotations(Actor self, WorldRenderer wr)
+		{
+			var map = world.Map;
+			var plan = tool.Plan;
+			if (plan == null)
+				yield break;
+
+			var mainColor = tool.EditorBlitSource != null ? Color.Cyan : Color.Red;
+
+			var points = plan.Points();
+			for (var i = 1; i < points.Length; i++)
+			{
+				yield return new CircleAnnotationRenderable(
+					map.CenterOfCell(points[i]), new WDist(128), 1, Color.Yellow, false);
```

Comment:
I will move the path planning's grid alignment to cell corners instead of cell centers. This is in practice how all the existing segments work, and how I think any future segments will work in practice. This will remove weird offsets.